### PR TITLE
[stdlib] Thou shalt not call objc_msgSend() without a function pointer cast.

### PR DESCRIPTION
**Pull request by gparker42**
Fixes test stdlib/ErrorBridged.swift on arm64.

rdar://30900987

### CODE REVIEW

**Comment by gparker42** [comment]
@swift-ci please smoke test and merge

**Comment by jrose-apple** [comment]
Oof. Can we pass `-D OBJC_OLD_DISPATCH_PROTOTYPES=0` when building the runtime (and possibly overlays) so this doesn't happen again?

File: stdlib/public/SwiftShims/NSErrorShims.h [hunk 1]
```diff
@@ -15,7 +15,9 @@
 NS_BEGIN_DECLS
 
 NS_INLINE void __NSErrorPerformRecoverySelector(_Nullable id delegate, SEL selector, BOOL success, void *_Nullable contextInfo) {
-    objc_msgSend(delegate, selector, success, contextInfo);
+    void (*msg)(_Nullable id, SEL, BOOL, void* _Nullable) =
+        (void(*)(_Nullable id, SEL, BOOL, void* _Nullable))objc_msgSend;
+    msg(delegate, selector, success, contextInfo);
```
Comment (dabrahams) [inline stdlib/public/SwiftShims/NSErrorShims.h]: Looks pretty subtle to me.  Doesn't this change deserve an explanatory comment in the code?

Comment (dabrahams) [inline stdlib/public/SwiftShims/NSErrorShims.h]: Oh, and *THANK YOU* for tracking this down!

Comment (gparker42) [inline stdlib/public/SwiftShims/NSErrorShims.h]: No comment needed. We'll enable compile-time enforcement of this soon.

Comment (gparker42) [inline stdlib/public/SwiftShims/NSErrorShims.h]: #8085

**Pull request merged**